modules/configuration/src/main/java/org/apache/ignite/internal/configuration/hocon/HoconListConfigurationSource.java

```diff
@@ -105,6 +112,12 @@ public void descend(ConstructableTreeNode node) {
 
         String syntheticKeyName = ((NamedListNode<?>) node).syntheticKeyName();
 
+        for (Pattern deletedPrefix : deletedPrefixes) {
+            if (deletedPrefix.matcher(join(appendKey(path, syntheticKeyName))).matches()) {
```

Comment:
Please at least call `join` once before the loop. This code is kind of inefficient, I don't like it.

I'm not sure we should spend time optimizing it right now, but it just bothers me, you know